cmd/noisd/genesis.go

```diff
@@ -0,0 +1,381 @@
+	genParams.MintParams.MintDenom = BaseCoinUnit
+	//  default mint params change accordingly
+	genParams.MintParams.InflationRateChange = sdk.NewDecWithPrec(13, 2)
+	genParams.MintParams.InflationMax = sdk.NewDecWithPrec(20, 2)
+	genParams.MintParams.InflationMin = sdk.NewDecWithPrec(7, 2)
+	genParams.MintParams.GoalBonded = sdk.NewDecWithPrec(67, 2)
+	genParams.MintParams.BlocksPerYear = uint64(60 * 60 * 8766 / 5) // assuming 5 second block time
+
+	// staking
+	genParams.StakingParams = stakingtypes.DefaultParams()
+	genParams.StakingParams.UnbondingTime = time.Hour * 24 * 7 * 2 // 2 weeks
+	genParams.StakingParams.MaxValidators = 45
+	genParams.StakingParams.BondDenom = genParams.NativeCoinMetadatas[0].Base
+
+	genParams.DistributionParams = distributiontypes.DefaultParams()
+	genParams.DistributionParams.BaseProposerReward = sdk.MustNewDecFromStr("0.01")
+	genParams.DistributionParams.BonusProposerReward = sdk.MustNewDecFromStr("0.04")
+	// according to tokenomics this should be 0
+	genParams.DistributionParams.CommunityTax = sdk.ZeroDec()
+	genParams.DistributionParams.WithdrawAddrEnabled = true
+
+	genParams.GovParams = govtypes.DefaultParams()
+	genParams.GovParams.DepositParams.MaxDepositPeriod = time.Hour * 24 * 14 // 2 weeks
+	genParams.GovParams.DepositParams.MinDeposit = sdk.NewCoins(sdk.NewCoin(
+		genParams.NativeCoinMetadatas[0].Base,
+		sdk.NewInt(1_000_000_000),
+	))
+	genParams.GovParams.TallyParams.Quorum = sdk.MustNewDecFromStr("0.2") // 20%
+	genParams.GovParams.VotingParams.VotingPeriod = time.Hour * 24 * 3    // 3 days
+
+	// crisis make this  a high fee
+	genParams.CrisisConstantFee = sdk.NewCoin(
+		genParams.NativeCoinMetadatas[0].Base,
+		sdk.NewInt(100_000_000_000),
+	)
+
+	genParams.SlashingParams = slashingtypes.DefaultParams()
+	genParams.SlashingParams.SignedBlocksWindow = int64(25000)                       // ~41 hr at 6 second blocks
+	genParams.SlashingParams.MinSignedPerWindow = sdk.MustNewDecFromStr("0.05")      // 5% minimum liveness
+	genParams.SlashingParams.DowntimeJailDuration = time.Minute                      // 1 minute jail period
+	genParams.SlashingParams.SlashFractionDoubleSign = sdk.MustNewDecFromStr("0.05") // 5% double sign slashing
+	genParams.SlashingParams.SlashFractionDowntime = sdk.MustNewDecFromStr("0.0001") // 0.01% liveness slashing
+
+	genParams.ConsensusParams = tmtypes.DefaultConsensusParams()
+	genParams.ConsensusParams.Block.MaxBytes = 21 * 1024 * 1024 // 21 MB max block size the default
+	genParams.ConsensusParams.Block.MaxGas = 4_500_000
+	genParams.ConsensusParams.Evidence.MaxAgeDuration = genParams.StakingParams.UnbondingTime
+	genParams.ConsensusParams.Evidence.MaxAgeNumBlocks = int64(genParams.StakingParams.UnbondingTime.Seconds()) / 3
+
+	genParams.WasmParams = wasmtypes.DefaultParams()
+	genParams.WasmParams.CodeUploadAccess = wasmtypes.AllowEverybody
+	genParams.WasmParams.InstantiateDefaultPermission = wasmtypes.AccessTypeEverybody
```

Comment:
Have we decided on these permissions?

Reply:
You can also set it to only addresses for upload  and then switch it to nobody through gov proposal